[jit] Implement one-element MemoryDAG::mayContainAlias more efficiently

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66295
* #65344
* __->__ #65178

There is no need to copy the MemoryLocations in this case.

Differential Revision: [D30984442](https://our.internmc.facebook.com/intern/diff/D30984442/)

swolchok committed:
[jit] Implement one-element MemoryDAG::mayContainAlias more efficiently

There is no need to copy the MemoryLocations in this case.

Differential Revision: [D30984442](https://our.internmc.facebook.com/intern/diff/D30984442/)

[ghstack-poisoned]

eellison commented on torch/csrc/jit/passes/utils/memory_dag.cpp:
```diff
@@ -100,13 +110,8 @@ void MemoryDAG::collectAllContainedMemoryLocationsImpl(
 }
 
 bool MemoryDAG::mayContainAliasImpl(const Element* a, const Element* b) const {
-  MemoryLocations all_a_mlocs;
```
is this just a formatting change ? (for my own understanding)

swolchok replied:
no.`getAllContainedMemoryLocations()` is more efficient than `collectAllContainedMemoryLocations(x)` because it can return a const reference to the cached value if it exists rather than copying the cached value into x.

eellison reviewed (commented):
looks fine to me but had some questions about the caching in the other pr first

swolchok committed:
Update on "[jit] Implement one-element MemoryDAG::mayContainAlias more efficiently"


There is no need to copy the MemoryLocations in this case.

Differential Revision: [D30984442](https://our.internmc.facebook.com/intern/diff/D30984442/)

[ghstack-poisoned]

swolchok committed:
Update on "[jit] Implement one-element MemoryDAG::mayContainAlias more efficiently"


There is no need to copy the MemoryLocations in this case.

Differential Revision: [D30984442](https://our.internmc.facebook.com/intern/diff/D30984442/)

[ghstack-poisoned]

swolchok commented:
ping

swolchok committed:
Update on "[jit] Implement one-element MemoryDAG::mayContainAlias more efficiently"


There is no need to copy the MemoryLocations in this case.

Differential Revision: [D30984442](https://our.internmc.facebook.com/intern/diff/D30984442/)

[ghstack-poisoned]

swolchok committed:
Update on "[jit] Implement one-element MemoryDAG::mayContainAlias more efficiently"


There is no need to copy the MemoryLocations in this case.

Differential Revision: [D30984442](https://our.internmc.facebook.com/intern/diff/D30984442/)

[ghstack-poisoned]

swolchok committed:
Update on "[jit] Implement one-element MemoryDAG::mayContainAlias more efficiently"


There is no need to copy the MemoryLocations in this case.

Differential Revision: [D30984442](https://our.internmc.facebook.com/intern/diff/D30984442/)

[ghstack-poisoned]

swolchok committed:
Update on "[jit] Implement one-element MemoryDAG::mayContainAlias more efficiently"


There is no need to copy the MemoryLocations in this case.

Differential Revision: [D30984442](https://our.internmc.facebook.com/intern/diff/D30984442/)

[ghstack-poisoned]

swolchok committed:
Update on "[jit] Implement one-element MemoryDAG::mayContainAlias more efficiently"


There is no need to copy the MemoryLocations in this case.

Differential Revision: [D30984442](https://our.internmc.facebook.com/intern/diff/D30984442/)

[ghstack-poisoned]
